### PR TITLE
BG-18418 Fix offline signing

### PR DESCRIPTION
**Pull request by argjv**
Add methods to produce a unsigned serialized transaction and to sign it in an offline environment

### CODE REVIEW

File: package-lock.json [hunk 1]
```diff
@@ -3223,28 +3223,28 @@
       "dependencies": {
         "abbrev": {
           "version": "1.1.1",
-          "resolved": "https://registry.npmjs.org/abbrev/-/abbrev-1.1.1.tgz",
+          "resolved": false,
```
[the remark this answers is not in the document]
Comment (argjv) [reply package-lock.json]: No idea. This is after running `npm audit fix` due to some new issues.

File: src/coin/xtz/transactionBuilder.ts [hunk 2]
```diff
@@ -294,18 +266,43 @@ export class TransactionBuilder extends BaseTransactionBuilder {
...
   }
+  // endregion
+
+  // region AddressInitialization builder methods
+  /**
+   * The public key to reveal.
+   *
+   * @param {string} publicKey A Tezos public key
+   */
+  publicKey(publicKey: string): void {
```
[the remark this answers is not in the document]
Comment (argjv) [reply src/coin/xtz/transactionBuilder.ts]: This is used in reveal operations to specify the pub key to reveal

File: package-lock.json (hunk 1 above)
[the remark this answers is not in the document]
Comment (CapnMigraine) [reply package-lock.json]: Ok, I see some other differences when I re-create `node_modules` and `package-lock.json`. Make sure we're merging the right one

File: src/coin/xtz/transactionBuilder.ts (hunk 2 above)
[the remark this answers is not in the document]
Comment (CapnMigraine) [reply src/coin/xtz/transactionBuilder.ts]: I see, ty

**Pull request merged**